torch/csrc/jit/codegen/cuda/kernel.cpp

```diff
@@ -155,6 +155,139 @@ std::pair<std::string, std::string> codeGeneration(Fusion& fusion) {
   return std::make_pair(func_name, str_stream.str());
 };
 
+bool validateKernelArgTensor(
+    const at::Tensor& arg,
+    const Val* param,
+    int device_index,
+    std::stringstream& msg) {
+  // Arg is a tensor. Param must be a tensor too.
+  if (*param->getValType() != ValType::TensorView) {
+    msg << "Argument is a tensor, but the parameter is not.";
+    return false;
+  }
+  // Check the rank of the tensors.
+  size_t arg_dim = arg.dim();
+  const auto& root_dom = *static_cast<const TensorView*>(param)->getRootDomain();
+  size_t param_dim = 0;
+  for (size_t i = 0; i < root_dom.nDims(); ++i) {
+    if (!root_dom.axis(i)->isReduction())
+      ++param_dim;
+  }
+  if (arg_dim != param_dim) {
+    msg << "Argument tensor's rank is " << arg_dim << ", but the parameter is "
+        << param_dim;
+    return false;
+  }
+  if (arg.device().index() != device_index) {
+    msg << "Argument is on device that is not compiled for";
+    return false;
+  }
+  // Check element type
+  at::ScalarType arg_data_type = arg.scalar_type();
+  DataType param_data_type = *param->getDataType();
+  bool match = false;
+  switch (arg_data_type) {
+    case at::ScalarType::Int:
+      match = param_data_type == DataType::Int;
```

Comment:
We don't support `at::ScalarType::Int` yet. We should error out here.

`DataType::Int` in our codegen is mapped to `size_t`, if you look at our `type.cpp`. at::ScalarType::Int is mapped to `int`, which is `int32_t` (You can check out `c10/core/ScalarType.h`).

Reply:
Are you sure? I was pretty confident they were using `int64_t`

Reply:
`at::ScalarType::Int` appears to be defined as `int` in `c10/core/ScalarType.h`.

Reply:
@jjsjann123 In `type.cpp`, we have this:
```
165 static _enum_unordered_map<at::ScalarType, DataType> at_type_map{
 166     {at::ScalarType::Bool, DataType::Bool},
 167     {at::ScalarType::Float, DataType::Float},
 168     {at::ScalarType::Half, DataType::Half},
 169     {at::ScalarType::Int, DataType::Int}};
```
Doesn't this mean `ScalarType::Int` corresponds to `DataType::Int`? Yes, in the same file, there is also:

```
 43 static _enum_unordered_map<DataType, std::string> data_type_string_map{
 44     {DataType::Bool, "bool"},
 45     {DataType::Float, "float"},
 46     {DataType::Half, "__half"},
 47     {DataType::Int, "size_t"}};
```

I'm not sure how these maps are used, but they don't seem correct unless `at::ScalarType::Int` is defined as `size_t`, which doesn't seem to be the case.